tests/run.sh

```diff
@@ -23,7 +23,7 @@ fi
 # Change to file's dir (POSIXly).
 cd -P -- "$(dirname -- "$0")"
 this_dir=$PWD
-source_dir="$PWD/.."
+source_dir="${this_dir%/*}"
```

Comment:
This assumes that `$this_dir` (thus `$PWD`) contains a slash (and it was not all that easy to figure out what this does). I'm not really sure if this always works correctly...?